python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -431,16 +536,45 @@ def __arrow_ext_class__(self):
         """
         return ArrowVariableShapedTensorArray
 
+    if _arrow_extension_scalars_are_subclassable():
+        # TODO(Clark): Remove this version guard once we only support Arrow 9.0.0+.
+        def __arrow_ext_scalar_class__(self):
+            """
+            ExtensionScalar subclass with custom logic for this array of tensors type.
+            """
+            return ArrowTensorScalar
```

Comment:
Now that we have `ArrowTensorScalar.as_py()` delegating to `self.type._extension_scalar_to_ndarray()`, we can use the same extension scalar type for both `ArrowTensorArray` and `ArrowVariableShapedTensorArray`.